File: common/Tribe.ts
import PairingRule from './PairingRule'

interface Tribe {
    id: string;
    name: string;
    badgesEnabled?: boolean;
    callSignsEnabled?: boolean;
    defaultBadgeName?: string;
    alternateBadgeName?: string;
    email?: string;
    pairingRule?: PairingRule
}

export default Tribe

File: coupling-libraries/model/src/commonMain/kotlin/com/zegreatrob/coupling/model/tribe/Tribe.kt
package com.zegreatrob.coupling.model.tribe

data class Tribe(
    val id: TribeId,
    val pairingRule: PairingRule = PairingRule.LongestTime,
    val badgesEnabled: Boolean = false,
    val defaultBadgeName: String = "Default",
    val alternateBadgeName: String = "Alternate",
    val email: String? = null,
    val name: String? = null,
    val callSignsEnabled: Boolean = false,
    val animationEnabled: Boolean = true,
    val animationSpeed: Double = 1.0,
    val imageURL: String? = null,
)

val defaultTribe = Tribe(TribeId("DEFAULT"))
